flink-runtime/src/main/java/org/apache/flink/runtime/state/IncrementalRemoteKeyedStateHandle.java

```diff
@@ -324,7 +325,7 @@ public void registerSharedStates(SharedStateRegistry stateRegistry, long checkpo
         for (Map.Entry<StateHandleID, StreamStateHandle> sharedStateHandle :
                 sharedState.entrySet()) {
             SharedStateRegistryKey registryKey =
-                    createSharedStateRegistryKeyFromFileName(sharedStateHandle.getKey());
+                    createSharedStateRegistryKey(sharedStateHandle.getValue());
```

Comment:
With this change, the key in `IncrementalRemoteKeyedStateHandle.sharedState` is not actually a key anymore, but just a local file name used on recovery.

So keeping it as is seems confusing to me.

I think it would be clearer if we'd keep that property that SharedStateRegistry key is the same key as in this map. That would require storing the local file path explicitly (which is good), for example by adding a holder class for `StreamStateHandle` and the local path.
WDYT?

Reply:
I agree, but that need modify the checkpoint metadata serializer, what do you think about opening a new ticket to do this? If possible, I will start as soon as this ticket is completed.

Reply:
I rather see it as a part of this ticket because it's directly related and one should NOT be merged without the other.
Why would you implement this as a separate ticket?

Reply:
I thought it was just a structural improvement before, but now I realize that it was really ill-conceived . We shouldn't be "Pollution first, governance later". Thanks for the correction @rkhachatryan .